### PR TITLE
Add cloud-init config to build rancher image

### PR DESCRIPTION
**Pull request by jordojordo**
This adds the functionality for [cloud-init](https://cloudinit.readthedocs.io/en/latest/) to build a unique Rancher image from a supplied repo and branch.

The workflow is like this:
1. Supply droplet name, repo, branch, access token, ssh fingerprint, and password
2. `./cloud-config` is generated with the supplied repo and branch
3. DigitalOcean spins up the droplet
4. Once the droplet is ready cloud-init starts it's service
- cloud-init will install git and docker
- then downloads the Dockerfile from this repo
- clones the supplied repo with a specified branch
- builds the dashboard image
- runs the newly created image with a dashboard from the test branch

### CODE REVIEW

File: cloud-config-tmp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+groups:
+  - docker
+
+system_info:
+  default_user:
+    groups: [docker]
+
+runcmd:
+  - "mkdir -p /run/dockerfiles"
+  - "wget https://raw.githubusercontent.com/jordojordo/digital-ocean-droplet-rancher/main/Dockerfile -P /run/dockerfiles/"
```
Comment (jordojordo) [inline cloud-config-tmp:19]: This will need to be the upstream url, was just using my fork for testing

File: .gitignore [hunk 2]
```diff
@@ -13,3 +13,6 @@
 
 # Dependency directories (remove the comment below to include it)
 # vendor/
+
+# cloud-config generated from build
+cloud-config
```
Comment (rak-phillip) [inline .gitignore:18]: Can we fix these `No newline at end of file` warnings throughout?

**Review by rak-phillip — APPROVED** [review]
Very excited to see this working, this is great stuff 😁

Everything looks good to me, I only ask that we clean up the `No newline at end of file` warnings throughout.

**Review by rak-phillip — APPROVED** [review]
Looks good 🙂

**Comment by jordojordo** [comment]
I don't have permission to merge, so it's up to you!

**Pull request merged**